test/unit/services/segment_delete_service_test.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class SegmentDeleteServiceTest < ActiveSupport::TestCase
+  def setup
+    @token = 'example-token'
+    @user = FactoryBot.build_stubbed(:admin)
+    @segment_delete_service = SegmentDeleteService.new(user.id)
+  end
+
+  attr_reader :user, :segment_delete_service, :token
+
+  class DeleteUserRequestTest < SegmentDeleteServiceTest
+    def setup
+      super
+      segment_delete_service.stubs(token: token)
+    end
+
+    test '#delete_user! does the right request to delete' do
+      request = delete_request(status: 200)
+
+      assert segment_delete_service.delete_user!
+      assert_requested request
+    end
+
+    test '#delete_user and #delete_user! does the right request to delete' do
```

Comment:
We don't need to repeat "#delete_user!". This test is only about "#delete_user". The rest is abstraction.